packages/cli/src/specs/__snapshots__/repo-builder.spec.ts.snap

```diff
@@ -1,5 +1,66 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`Repo Builder all in a class: all in a class 1`] = `
+"import { Store, createState, withProps } from '@ngneat/elf';
+import { withEntities, selectAll, deleteEntities, addEntities, updateEntities, withUIEntities, withActiveId, selectActiveEntity, setActiveId, withActiveIds, selectActiveEntities, toggleActiveIds } from '@ngneat/elf-entities';
+import { Observable } from 'rxjs';
+import { withRequestsCache, withRequestsStatus } from '@ngneat/elf-requests';
+
+export interface TodoUI {
+  id: number;
+}
+
+export interface Todo {
+  id: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface TodosProps {
+}
+
+export class TodosRepository {
+  activeTodos$: Observable<Todo[]>;
+  activeTodo$: Observable<Todo | undefined>;
+  todos$: Observable<Todo[]>;
+
+  private store: ReturnType<TodosRepository['createStore']>;
```

Comment:
I think it'll be better to use real private fields:

```ts
#store: ReturnType<TodosRepository['createStore']>;

 #createStore() {
  ...
}
```

Reply:
I'm wondering about your frequency of usage of real private fields. I haven't seen them at all.

Reply:
They are supported in all major browsers. If we use it, no one can do something like `(repo as any).store`.

Reply:
in this case, there are 2 problems
1. when we use native private `#store` then a property `#store` stops interring a type.
2. we can't set a return type of the method `ReturnType<this['#createStore']>;`

Reply:
Ok, but we should mention this in the docs. Maybe users will change it to `#` and complain about what you're saying.

Reply:
@NetanelBasal I have a thought. 
We can use native private fields without explicitly setting a type. In this case, IDE will not show a type hint on hover but all types will be correct.

If for the query properties there should be explicit type(to not scroll down to understand a type) that for store an explicit type is not so important.

<img width="343" alt="image" src="https://user-images.githubusercontent.com/31760529/154931670-3febbf4a-7751-41eb-83eb-00be216d784e.png">
<img width="697" alt="image" src="https://user-images.githubusercontent.com/31760529/154931716-b2202a2c-4429-4569-bb87-e95084e3dd4d.png">
<img width="833" alt="image" src="https://user-images.githubusercontent.com/31760529/154931749-722ac34f-21ff-4944-9ff6-b31e2f761b7f.png">

Reply:
As long as TS works as expected I'm fine

Reply:
could we add an option for native or ts `private`?

Reply:
I know by myself that native private will conflict with some eslint rules. For example 
<img width="767" alt="image" src="https://user-images.githubusercontent.com/31760529/154935823-0f7c7bf8-e791-4b89-acc4-ea9ba31466d9.png">

Reply:
Let's skip it for now.